packages/terra-table/src/SelectableUtils.js

```diff
@@ -16,21 +29,77 @@ const initialSingleSelectRowIndex = (rows) => {
 };
 
 /**
-* Returns if the selected row index is already selected for SingleSelectableRows.
-*/
+ * Returns the indexes of the selected rows for MultiSelectableRows.
+ * To be used in the constructor, to set initial state.
+ */
+const initialMultiSelectRowIndexes = (rows, maxSelectionCount) => {
+  if (!rows || !rows.length) {
+    return [];
+  }
+  // Find the rows which are selected and are selectable
+  const selectedIndexes = [];
+  const validatedMaxSelectionCount = validatedMaxCountSelection(rows, maxSelectionCount);
+  for (let i = 0; i < rows.length; i += 1) {
+    if (rows[i].props.isSelected) {
+      selectedIndexes.push(i);
+
+      if (selectedIndexes.length >= validatedMaxSelectionCount) {
+        break;
+      }
+    }
+  }
+  return selectedIndexes;
+};
+
+/**
+ * Returns a new array, updated with the newIndex being added or removed from the existing.
```

Comment:
nit: returns a new array _of the selected indexes_...

Reply:
Updated